pkg/prometheus/promcfg.go

```diff
@@ -813,7 +811,7 @@ func (cg *ConfigGenerator) generatePodMonitorConfig(
 	}...)
 
 	// Relabel targetLabels from Pod onto target.
-	for _, l := range m.Spec.PodTargetLabels {
+	for _, l := range append(m.Spec.PodTargetLabels, cg.spec.GlobalPodTargetLabels...) {
```

Comment:
Accidentally or not, ServiceMonitor/PodMonitor objects could remove global target labels by adding drop actions to their `spec.relabelConfigs`. I wonder whether we want the global target labels relabeling to happen last?

Reply:
I went back and forth on that one, what would be least surprising. But if we consider this a policy thing (which is actually my use-case, so I guess I should be on this side) then developers shouldn't be able to drop these.

Good catch!

Reply:
Wondering if this is a bit trickier than I initially thought? If the user would set a drop on relabelConfigs (or metricRelabelConfigs?), the order won't really matter, will it? We'll need to either reject monitors which try to touch these labels, or remove those relabelings, right? Or am I missing something obvious?

Reply:
@simonpasquier What do you think, is either of my suggestions reasonable? Or am I missing something?

Reply:
You're 100% correct. And IMHO it's quite impossible for the operator to ensure that a user relabeling configuration doesn't mutate/drop the metadata pod labels. For instance, adding this snippet to a service monitor's relabel config would prevent any attempt to enforce the  `foo` label:

```
- action: labeldrop
  regex: "__[m]eta_kubernetes_pod_(label)?_[f][o][o]"
- action: labeldrop
  regex: "[f][o][o]"
```

Reply:
Right. Should we just document this as a limitation?

Reply:
Yes I think it's a reasonable trade-off. But maybe `enforced...`  is a bit misleading and we need to stick with `podTargetLabels`?

Reply:
:+1:

Reply:
Removed the `enforced` bit from the name now